### PR TITLE
[spirv] use presumed loc for line info

### LINKED ISSUE #2333: [SPIRV] SPIR-V OpLine Line Numbers Off By One
The line numbers in the OpLine operations in the SPIR-V generated by dxc for the attached shader are off by 1, specificially 1 larger than they should be.

FWIW, I am guessing its because dxc is using the preprocessed version of the HLSL which has a #line inserted at the top which increases all original line numbers by one.

Here is the dxc command line which generates the bad SPIR-V:

dxc.exe -spirv -E MainPs -T ps_6_0 -fspv-debug=line -fspv-target-env=vulkan1.1 -Od -Zi hlsl.pp.line2.frag -Fo hlsl.pp.line2.frag.spv

The HLSL is attached. Remove the .txt suffix before compiling.

[hlsl.pp.line2.frag.txt](https://github.com/microsoft/DirectXShaderCompiler/files/3383949/hlsl.pp.line2.frag.txt)

### PR DESCRIPTION
**Pull request by jaebaek**
Fixes #2333

### CODE REVIEW

**Comment by jaebaek** [comment]
@ehsannas This failure is caused by `ninja: command not found` and I guess it might be just a temporal error and it will pass if we test it again.

I have no idea how to run the travis test again. Could you please help me?

**Comment by ehsannas** [comment]
@jaebaek Thanks. I have restarted the Travis bots

File: tools/clang/include/clang/SPIRV/SpirvBuilder.h [hunk 1]
```diff
@@ -618,21 +618,28 @@ SpirvBuilder::setDebugSource(uint32_t major, uint32_t minor,
...
-    fileStrings.push_back(
+    SpirvString *fileString =
         name.empty() ? nullptr
-                     : new (context) SpirvString(/*SourceLocation*/ {}, name));
+                     : new (context) SpirvString(/*SourceLocation*/ {}, name);
+    SpirvSource *debugSource = new (context)
+        SpirvSource(/*SourceLocation*/ {}, spv::SourceLanguage::HLSL, version,
+                    fileString, content);
+    module->addDebugSource(debugSource);
+    if (!mainSource)
```
Comment (ehsannas) [inline tools/clang/include/clang/SPIRV/SpirvBuilder.h]: This condition is always false!

Comment (ehsannas) [reply tools/clang/include/clang/SPIRV/SpirvBuilder.h]: Oops. I got it backwards.

**Pull request merged**